src/shared/term-before-position.ts

```diff
@@ -0,0 +1,12 @@
+/**
+ * Given a text, this returns the "word" right before a specific position.
+ *
+ * In this context, a word is anything between a space or newline, and provided
+ * position.
+ */
+export function termBeforePosition(text: string, position: number): string {
+  const textBeforePosition = text.slice(0, position).replace(/\n/g, ' ');
+  const lastSpaceIndex = textBeforePosition.lastIndexOf(' ');
+
+  return textBeforePosition.slice(lastSpaceIndex + 1);
```

Comment:
I think this function is generic enough that it can live in src/shared. It doesn't reference any sidebar-specific concept, it's just a text manipulation function of some sorts.

If other code can eventually benefit from it it's already there to "grab", without having to move it somewhere else at that point.

There are other functions in there which are also used only in one place.

Reply:
> I think this function is generic enough that it can live in src/shared. It doesn't reference any sidebar-specific concept, it's just a text manipulation function of some sorts.

Maybe. It depends on how exactly the rules for deciding where an at-mention is evolve over time. They might end up being more context-aware (eg. don't show suggestions if "@" appears in HTML attributes).

Reply:
What I meant is that this function is implemented in a way that it's not really coupled with at-mentions per se. There could be other use cases for it.

Reply:
Oh, wait. I misread your comment.

So basically you mean it is likely this function needs to evolve into a more sidebar-related direction. Yeah, I think that's a good point. I'll put it in one of the src/sidebar modules you suggested.